fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1897,15 +1897,8 @@ public int getAsInt() {
                 AddFieldClause addFieldClause = (AddFieldClause) alterClause;
                 modifyFieldColumns = Set.of(addFieldClause.getColName());
                 checkModifiedColumWithMaterializedViews(olapTable, modifyFieldColumns);
-
-                Locker locker = new Locker();
-                locker.lockDatabase(db, LockType.READ);
-                int id = 0;
-                try {
-                    id = olapTable.incAndGetMaxColUniqueId();
-                } finally {
-                    locker.unLockDatabase(db, LockType.READ);
-                }
+  
+                int id = colUniqueIdSupplier.getAsInt();
```

Comment:
If there are two statements running concurrently, how can you guarantee that they will get the unique ID? It seems that probably they will share the same IDs.

Reply:
> If there are two statements running concurrently, how can you guarantee that they will get the unique ID? It seems that probably they will share the same IDs.

We can not running two alter job in the same table right now, so there will be no concurrency issues.

Reply:
How is it guaranteed?

Reply:
> How is it guaranteed?

If one table has unfinished alter job, we can not submit a new alter job.